src/components/Users/CreateUserForm.tsx

```diff
@@ -107,14 +114,52 @@ export default function CreateUserForm({ onSubmitSuccess }: Props) {
   });
 
   const userType = form.watch("user_type");
+  const usernameInput = form.watch("username");
   const phoneNumber = form.watch("phone_number");
   const isWhatsApp = form.watch("phone_number_is_whatsapp");
 
   useEffect(() => {
     if (isWhatsApp) {
       form.setValue("alt_phone_number", phoneNumber);
     }
-  }, [phoneNumber, isWhatsApp, form]);
+    if (usernameInput && usernameInput.length > 0) {
+      form.trigger("username");
+    }
+  }, [phoneNumber, isWhatsApp, form, usernameInput]);
+
+  const { error, isLoading } = useQuery({
+    queryKey: ["checkUsername", usernameInput],
+    queryFn: query(userApi.checkUsername, {
+      pathParams: { username: usernameInput },
+      silent: true,
+    }),
+    enabled: usernameInput?.length >= 4,
+  });
+
+  const renderUsernameFeedback = (usernameInput: string) => {
+    if (
+      form.formState.errors.username &&
+      form.formState.errors.username?.message
+    ) {
+      return validateRule(false, form.formState.errors.username.message);
+    } else if (isLoading) {
+      return (
+        <div className="flex items-center gap-1">
+          <CareIcon
+            icon="l-spinner"
+            className="text-xl text-gray-500 animate-spin"
+          />
+          <span className="text-gray-500 text-sm">
+            {t("checking_availability")}
+          </span>
+        </div>
+      );
+    } else if (error) {
+      return validateRule(false, <>{t("username_not_available")}</>);
+    } else if (usernameInput && !form.formState.errors.username && !isLoading) {
```

Comment:
Shouldn't need to check !isLoading since you are already checking whether isLoading is true earlier.